File: CoreModels/Vehicle.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CoreModels
{
    public class Vehicle
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public VehicleType VehicleType { get; set; }
        public Status Status { get; set; }

    }
}

File: Admin/ClientApp/app/components/Vehicle/Vehicle.ts
export class Vehicle {
    id: number;
    name: string;
    vehicleType: {
        id: number;
        name: string;
        vehicles: any[];
    };
    status: {
        id: number;
        code: string;
        meaning: string;
        vehicles: any[];
    };
}
